core/server/api_container/server/startosis_engine/recipe/extractor.go

```diff
@@ -0,0 +1,78 @@
+package recipe
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/itchyny/gojq"
+	"github.com/kurtosis-tech/stacktrace"
+	"github.com/sirupsen/logrus"
+	"go.starlark.net/starlark"
+)
+
+func Extractor(query string, input []byte) (starlark.Comparable, error) {
+	logrus.Debugf("Running extractor against query '%v' and input '%v'", input, query)
+	jqQuery, err := gojq.Parse(query)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred when parsing field extractor '%v'", query)
+	}
+	var jsonBody interface{}
+	err = json.Unmarshal([]byte(input), &jsonBody)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred when parsing JSON response body:\n'%v'", jsonBody)
+	}
+	iter := jqQuery.Run(jsonBody)
+	parsedMatchList := []starlark.Value{}
+	for {
+		matchValue, ok := iter.Next()
+		if !ok {
+			break
+		}
+		logrus.Debugf("Found match '%v'", matchValue)
+		if err, ok := matchValue.(error); ok {
+			logrus.Warnf("Match '%v' was error type '%T'", err, err)
+			continue
+		}
+		if matchValue != nil {
+			logrus.Debug("Match was not an error, start parsing...")
+			parsedMatchValue := parseJsonValueToStarlark(matchValue)
+			logrus.Debugf("Parsed successfully from %v to %v", matchValue, parsedMatchValue)
```

Comment:
nit: since parseJsonValueToStarlark cannot throw an error, I am not sure it's valuable to log `Match was not an error, start parsing...` above, as you'll never eject up to this one

Reply:
Done